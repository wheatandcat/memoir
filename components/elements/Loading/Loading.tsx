import type { FC } from 'react';
import { ActivityIndicator } from 'react-native';
import View from 'components/atoms/View';
import theme from 'config/theme';

type Props = {
  size?: 'small' | 'large';
};

const Loading: FC<Props> = ({ size = 'large' }) => {
  return (
    <View m={4}>
      <ActivityIndicator
        size={size}
        color={theme().color.secondary.light}
        testID="atoms_loading"
      />
    </View>
  );
};

export default Loading;
