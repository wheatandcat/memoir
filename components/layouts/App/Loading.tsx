import ALoading from "@/components/elements/Loading";
import View from "@/components/elements/View";
import theme from "config/theme";
import { type FC, memo } from "react";
import { StyleSheet } from "react-native";

const Loading: FC = () => {
  return (
    <View style={styles.root}>
      <ALoading />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
    height: "100%",
  },
});

export default memo(Loading);
