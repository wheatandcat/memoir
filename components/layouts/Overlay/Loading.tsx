import Text from "@/components/elements/Text";
import View from "@/components/elements/View";
import theme from "@/config/theme";
import { type FC, memo, useEffect, useState } from "react";
import {
  ActivityIndicator,
  StyleSheet,
  useWindowDimensions,
} from "react-native";

export type Props = {
  text: string;
};

const Loading: FC<Props> = (props) => {
  const [count, setCount] = useState(0);
  const width = useWindowDimensions().width;

  useEffect(() => {
    const interval = setInterval(() => {
      setCount(count + 1);
    }, 200);
    return () => clearInterval(interval);
  }, [count]);

  const style = { paddingLeft: width / 2 - (props.text?.length || 0) * 14 };

  return (
    <View style={styles.root} testID="overlay-loading">
      <View style={[styles.loading, style]}>
        <View mt={1}>
          <ActivityIndicator color={theme().color.base.light} />
        </View>
        <View ml={2}>
          <Text
            color="baseLight"
            fontFamily="NotoSansJP-Bold"
            textAlign="center"
          >
            {props.text} {(() => {
              if (count % 3 === 0) {
                return ".";
              }
              if (count % 3 === 1) {
                return "..";
              }
              if (count % 3 === 2) {
                return "...";
              }

              return "";
            })()}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: `${theme().color.secondary.dark}99`,
    height: "100%",
    width: "100%",
    justifyContent: "center",
    position: "absolute",
    top: 0,
    left: 0,
  },
  loading: {
    flexDirection: "row",
    alignItems: "center",
  },
});

export default memo(Loading);
