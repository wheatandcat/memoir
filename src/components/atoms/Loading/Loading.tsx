import React from 'react';
import { ActivityIndicator } from 'react-native';
import View from 'components/atoms/View';
import theme from 'config/theme';

type Props = {
  size?: 'small' | 'large';
};

const Loading: React.FC<Props> = (props) => {
  return (
    <View>
      <ActivityIndicator
        size={props.size}
        color={theme().color.secondary.light}
      />
    </View>
  );
};

export default Loading;
