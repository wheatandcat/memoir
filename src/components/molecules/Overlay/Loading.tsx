import React, { memo, useEffect, useState } from 'react';
import {
  StyleSheet,
  ActivityIndicator,
  useWindowDimensions,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';

export type Props = {
  text: string;
};

const Loading: React.FC<Props> = (props) => {
  const [count, setCount] = useState(0);
  const width = useWindowDimensions().width;

  useEffect(() => {
    const interval = setInterval(() => {
      setCount(count + 1);
    }, 200);
    return () => clearInterval(interval);
  }, [count]);

  const style = { paddingLeft: width / 2 - 50 };

  return (
    <View style={styles.root}>
      <View style={[styles.loading, style]}>
        <View mt={1}>
          <ActivityIndicator color={theme().color.base.light} />
        </View>
        <View ml={2}>
          <Text color="baseLight" fontFamily="NotoSansJP-Bold">
            {props.text}{' '}
            {(() => {
              if (count % 3 === 0) {
                return '.';
              }
              if (count % 3 === 1) {
                return '..';
              }
              if (count % 3 === 2) {
                return '...';
              }

              return '';
            })()}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: `${theme().color.secondary.dark}99`,
    height: '100%',
    width: '100%',
    justifyContent: 'center',
    position: 'absolute',
    top: 0,
    left: 0,
  },
  loading: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export default memo(Loading);
