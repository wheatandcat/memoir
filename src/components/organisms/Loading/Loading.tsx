import React from 'react';
import { ActivityIndicator, Dimensions, StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import theme from 'config/theme';

const deviceHeight = Dimensions.get('screen').height;
const deviceWidth = Dimensions.get('window').width;

type Props = {
  loading: boolean;
  row?: boolean;
};

const Loading: React.FC<Props> = (props) => {
  if (!props.loading) return <>{props.children}</>;

  return props.row ? (
    <ActivityIndicator size="large" />
  ) : (
    <View style={[styles.container, styles.background]} {...props}>
      <View style={styles.inner}>
        <ActivityIndicator size="large" />
      </View>
    </View>
  );
};

export default Loading;

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: deviceWidth,
    height: deviceHeight,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    opacity: 0.8,
  },
  inner: {
    flex: 1,
    position: 'absolute',
    top: '35%',
  },
  background: {
    backgroundColor: theme().color.background.light,
    height: '100%',
    width: '100%',
  },
});
