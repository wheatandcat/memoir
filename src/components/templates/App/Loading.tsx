import React, { memo } from 'react';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import theme from 'config/theme';
import ALoading from 'components/atoms/Loading';

export type Props = {};

const Loading: React.FC<Props> = () => {
  return (
    <View style={styles.root}>
      <ALoading />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%',
  },
});

export default memo(Loading);
