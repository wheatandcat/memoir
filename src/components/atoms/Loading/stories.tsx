import React from 'react';
import { storiesOf } from '@storybook/react-native';
import View from 'components/atoms/View';
import Loading from './';

storiesOf('atoms', module).add(
  'Loading',
  () => (
    <View>
      <View p={3}>
        <Loading size="small" />
      </View>
      <View p={3}>
        <Loading />
      </View>
      <View p={3}>
        <Loading size="large" />
      </View>
    </View>
  ),
  { loki: { skip: true } }
);
