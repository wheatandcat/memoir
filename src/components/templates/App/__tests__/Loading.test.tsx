import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Loading, { Props } from '../Loading';

const propsData = (): Props => ({});

describe('components/templates/App/Loading.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Loading {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
