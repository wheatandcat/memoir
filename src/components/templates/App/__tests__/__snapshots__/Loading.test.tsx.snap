// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/templates/App/Loading.tsx 正常にrenderすること 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Loading
    size="large"
  />
</View>
`;
